app/components/comparison_table_component.rb

```diff
@@ -21,13 +23,31 @@ def initialize(report:, advice_page:, table_name:, index_params:, headers: [], c
     @advice_page_tab = advice_page_tab
   end
 
-  renders_many :rows, ->(**args) do
-    args[:advice_page] = @advice_page
-    args[:advice_page_tab] = @advice_page_tab
-    RowComponent.new(**args)
+  renders_many :rows, ->(**kwargs) do
+    kwargs[:advice_page] = @advice_page
+    kwargs[:advice_page_tab] = @advice_page_tab
+    RowComponent.new(**kwargs)
   end
 
-  renders_one :footer
+  renders_many :footnotes, 'ComparisonTableComponent::FootnoteComponent'
+  renders_many :notes, 'ComparisonTableComponent::NoteComponent'
+
+  def before_render
+    collect_references
+  end
+
+  def collect_references
+    seen = {}
+    rows.each do |row|
+      row.to_s # force early render to collect references, haven't found a better way as yet
```

Comment:
Presumably this causes a "double render" of the row, e.g. its rendered as a string here then later to HTML. I guess this could have performance issues for larger tables?

Was wondering if it'd be possible for the RowComponent or ReferenceComponent instances to call a method on the table component instance (or a shared object) so that the seen values are collected as the table is rendered?

Not suggesting a change now, but wondered if you'd tried that. We can merge this and then see if there are problems.

Reply:
Fairly sure it's only rendered once - presuming this is the relevant slot code anyway: https://github.com/ViewComponent/view_component/blob/0f953d95b43abdb48dccdac7f6afb157bb8773f6/lib/view_component/slot.rb#L45

But I'll have a proper look when I'm back in (or before then if I get chance). I couldn't work out how to access the table from the Row/Reference, but may have missed something.